lib/btree.js

```diff
@@ -0,0 +1,240 @@
+  const deletedData = element.data;
+  const leftChild = element.child;
+  const rightChild = node[elementIndex + 1].child;
+  if (leftChild.length > minDegree) {
+    const lowerLimit = extractLowerLimit(leftChild, minDegree);
+    element.value = lowerLimit.value;
+    element.data = lowerLimit.data;
+  } else if (rightChild.length > minDegree) {
+    const upperLimit = extractUpperLimit(rightChild, minDegree);
+    element.value = upperLimit.value;
+    element.data = upperLimit.data;
+  } else {
+    joinNodes(node, elementIndex, elementIndex + 1);
+    rightChild.splice(minDegree - 1, 1);
+  }
+  return deletedData;
+};
+
+class BTree {
+
+  constructor(degree = DEFAULT_DEGREE) {
+    this.root = [empty()];
+    this.minDegree = degree;
+  }
+
+  get(value) {
+    let currNode = this.root;
+    while (currNode) {
+      const [found, i] = binarySearch(currNode, value);
+      if (found) {
+        return currNode[i].data;
+      }
+      currNode = currNode[i].child;
+    }
+    return null;
+  }
+
+  insert(value, data, replace = false) {
+    const newElement = new Element(value, data);
+    if (this.root.length === 1) {
+      this.root.unshift(newElement);
+      return this;
+    }
+    if (this.root.length === this.minDegree * 2) {
+      this.root = [empty(this.root)];
+      splitNode(this.root, 0);
+    }
+    let currNode = this.root;
+    while (true) {
+      const [found, nextNodeIndex] = binarySearch(currNode, value);
+      if (!replace && found) {
+        throw new Error(`value ${value} already is in the tree`);
+      }
+      if (isLeaf(currNode)) {
+        currNode.splice(nextNodeIndex, 0, newElement);
+        return this;
+      }
+      let nextNode = currNode[nextNodeIndex].child;
+      if (nextNode.length === this.minDegree * 2) {
+        splitNode(currNode, nextNodeIndex);
+        const element = currNode[nextNodeIndex];
+        if (!replace && element.value === value) {
+          throw new Error(`value ${value} already is in the tree`);
+        }
+        if (element.value > value) {
+          nextNode = element.child;
+        }
+      }
+      currNode = nextNode;
+    }
+  }
+
+  iterator(start, finish) {
+    return inorderTraversal(start, finish, this.root);
+  }
+
+  remove(value) {
+    let currNode = this.root;
+    while (!isLeaf(currNode)) {
+      const [found, index] = binarySearch(currNode, value);
+      if (found) {
+        if (this.root.length === 1) this.root = this.root[0].child;
+        return deleteElement(currNode, index, this.minDegree);
+      } else {
+        let nextNode = currNode[index].child;
+        if (nextNode.length === this.minDegree) {
+          nextNode = growChild(currNode, index);
+        }
+        currNode = nextNode;
+      }
+    }
+    if (this.root.length === 1) this.root = this.root[0].child;
+    const [found, index] = binarySearch(currNode, value);
+    if (found) {
+      return currNode.splice(index, 1)[0].data;
+    } else return null;
```

Comment:
If the comment about removing this code is not applicable then you have to fix the formatting of this, we either have
* both branches with braces or
* both branches without braces